autopush/router/apnsrouter.py

```diff
@@ -27,78 +27,129 @@ class APNSRouter(object):
               255: 'Unknown',
               }
 
-    def _connect(self):
-        """Connect to APNS"""
-        self.apns = apns.APNs(use_sandbox=self.config.get("sandbox", False),
-                              cert_file=self.config.get("cert_file"),
-                              key_file=self.config.get("key_file"),
-                              enhanced=True)
+    def _connect(self, cert_info):
+        """Connect to APNS
+
+        :param cert_info: APNS certificate configuration info
+        :type cert_info: dict
+
+        returns an instance of APNs that can be stored under the proper
+        channel.
+
+        """
+        # Do I still need to call this in _error?
+        return apns.APNs(
+            use_sandbox=cert_info.get("sandbox", False),
+            cert_file=cert_info.get("cert"),
+            key_file=cert_info.get("key"),
+            enhanced=True)
 
     def __init__(self, ap_settings, router_conf):
         """Create a new APNS router and connect to APNS"""
         self.ap_settings = ap_settings
         self._base_tags = []
-        self.config = router_conf
-        self.default_title = router_conf.get("default_title", "SimplePush")
-        self.default_body = router_conf.get("default_body", "New Alert")
-        self._connect()
+        self.apns = dict()
+        self._config = router_conf
+        for channel in self._config.keys():
```

Comment:
the iOS team have certs assigned to the firefox channel release, so these would map to "firefox", "beta", "gecko" and whatever other channel definition they want to use (e.g. "nightly?")